browser/main/SideNav/StorageItem.js

```diff
@@ -131,8 +132,64 @@ class StorageItem extends React.Component {
     }
   }
 
+  handleDragEnter (e) {
+    e.dataTransfer.setData('defaultColor', e.target.style.backgroundColor)
+    e.target.style.backgroundColor = 'rgba(129, 130, 131, 0.08)'
+  }
+
+  handleDragLeave (e) {
+    e.target.style.opacity = '1'
+    e.target.style.backgroundColor = e.dataTransfer.getData('defaultColor')
+  }
+
+  handleDrop (e, storage, folder, dispatch, location) {
+    e.target.style.opacity = '1'
+    e.target.style.backgroundColor = e.dataTransfer.getData('defaultColor')
+    const noteData = JSON.parse(e.dataTransfer.getData('note'))
+    if (folder.key !== noteData.folder) {
+      dataApi
+       .createNote(storage.key, {
+         content: noteData.content,
+         createdAt: noteData.createdAt,
+         folder: folder.key,
+         isStarred: noteData.isStarred,
+         storage: storage,
+         title: noteData.title,
+         tags: noteData.tags,
+         type: noteData.type,
+         updatedAt: noteData.updatedAt,
+         description: noteData.description,
+         snippets: noteData.snippets
+       })
+       .then((note) => {
+         dispatch({
+           type: 'UPDATE_NOTE',
+           note: note
+         })
+         hashHistory.push({
+           pathname: location.pathname,
+           query: {key: note.storage + '-' + note.key}
+         })
+       })
+
```

Comment:
We need to handle rollback this copy process just in case it does not work. In this state, only to be copied the note without delete.